pkg/asset/machines/openstack/machines.go

```diff
@@ -86,10 +86,10 @@ func provider(clusterID string, platform *openstack.Platform, mpool *openstack.M
 			Kind:       "OpenstackProviderSpec",
 		},
 		Flavor: mpool.FlavorName,
-		/*RootVolume: openstackprovider.RootVolume{
-			VolumeType: pointer.StringPtr(mpool.Type),
-			Size:       pointer.Int64Ptr(int64(mpool.Size)),
-		},*/
+		RootVolume: &openstackprovider.RootVolume{
+			Size:       mpool.RootVolume.Size,
+			SourceType: mpool.RootVolume.Type,
```

Comment:
I doubt that `SourceType` is correct here.
Looking at https://github.com/gophercloud/gophercloud/blob/8f47ec3/openstack/compute/v2/extensions/bootfromvolume/requests.go#L81, we'd probably want `VolumeType` instead, however that doesn't seem implemented in CAPO?

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/7e4e4a8/api/v1alpha2/types.go#L112

Reply:
Source type should be "image", DeviceType is mpool.RootVolume.Type